src/mugs.js

```diff
@@ -124,6 +125,9 @@ define([
          */
         validate: function (attr) {
             var mug = this;
+
+            util.checkForFormSubmissions(mug.form);
```

Comment:
The motivation for this change is to show the warning as soon as a user submits their first form, since it is believed that people often do so via app preview while building the form.

We could rate limit the ajax requests to something small like 30 seconds.  Preventing concurrent requests also sounds like a reasonable approach.  Is there a common paradigm for rate limiting ajax requests?

Reply:
https://github.com/dimagi/Vellum/pull/953/commits/ff48c32eb316f686f4ec42c92e5bf69e51905ee1

Reply:
^ I'm assuming that checkForFormSubmissions is called within the same thread to limit concurrent requests.  If there are N threads, then there could be at most N concurrent ajax requests with my solution.

Reply:
I believe JS is always single-threaded (and async) unless you use some kind of worker process thing, which is generally lower performance for most things.

Reply:
Oh, https://github.com/dimagi/Vellum/commit/ff48c32eb316f686f4ec42c92e5bf69e51905ee1 doesn't prevent a storm of requests one after another. [`_.throttle`](https://underscorejs.org/#throttle) should do what you want.